spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -20,33 +20,141 @@
+=== Concepts
 
-In contrast, synchronous communication couples services together, binding their uptime, failure and handling of load to each other. In a chain of synchronous communication, the entire conversation will only be successful if all parties in the chain are responsive - if they are all running, processing messages successfully, and not overloaded. If just one party has a problem, all will effectively exhibit the same problem.
+This section describes the different concepts introduced by the Reactive Messaging specification
 
-In a microservice based architecture, temporal coupling results in a fragile system, with resilience and scaling properties that are worse than a monolith, hence, it is important for microservice based architectures to embrace asynchronous communication as much as possible. MicroProfile currently does not offer support for asynchronous messaging, only synchronous messaging using JAX-RS and MicroProfile REST client are supported.
+==== Overall architecture
 
-Java EE offers JMS and Message Driven Beans for handling asynchronous communication, however, there are some problems with these specifications:
+An application using Reactive Messaging is composed by CDI beans consuming, producing and processing _messages_.
 
-* Both are designed for a technology landscape where messaging was a heavy weight process used on the edge of system to hand control of a transaction from system to another, consequently there is a lot of boilerplate associated with using these technologies.
-* It is assumed their design that consistency is handled using distributed transactions. However, modern message brokers, popular in microservice deployments, such as Apache Kafka, Amazon Kinesis and Azure Event Hubs, do not support XA transactions, rather, message acknowledgement is handled using offsets with at least once delivery guarantees.
-* They do not have support for asynchronous IO, it is assumed that message processing is done on a single thread, however many modern specs are moving to asynchronous IO.
+These messages can be _internal_ messages and be received/forwared from/to different brokers.
 
-Hence a lighter weight, reactive solution to messaging is desirable for MicroProfile to ensure microservices written using MicroProfile are able to meet the demands required by the architecture.
+image::overall.png[Overall architecture]
 
-=== Design
+Application's beans contain methods annotated with `@Incoming` and `@Outgoing` annotations. 
+A method with an `@Incoming` annotation consumes messages from a _channel_. 
+A method with an `@Outgoing` annotation produces messages, it emits messages to a _channel_.
```

Comment:
I would use 'publishes' instead of 'emits'. Particularly given the underlying technology. Emit has more of the expelling pushing away to it or just doing rather than sending to a particular destination perhaps. 'sends' is anotehr possibility. (The sentance is clear in meaning as it is though.)

Reply:
Agreed!